pkg/gameserverallocations/controller_test.go

```diff
@@ -1157,6 +1158,86 @@ func TestMultiClusterAllocationFromRemote(t *testing.T) {
 			assert.Equal(t, expectedGSName, result.Status.GameServerName)
 		}
 	})
+	t.Run("Allocation is retried", func(t *testing.T) {
```

Comment:
There are tests for retry:
- https://github.com/googleforgames/agones/blob/master/pkg/gameserverallocations/controller_test.go#L985
- https://github.com/googleforgames/agones/blob/master/pkg/gameserverallocations/controller_test.go#L1082

You want to add a test that remote allocation does not respond for the specified timeout. For example set the timeout to 10ms and send a request to a non existing service or to a service which sleeps for 20ms before sending a response.

Reply:
I have added a test that sets the total timeout to 0 and so allows us to validate that total timeout affects the retries.